src/python/zquantum/core/interfaces/backend.py

```diff
@@ -128,25 +152,25 @@ def get_wavefunction(self, circuit: Circuit, **kwargs) -> Wavefunction:
             pyquil.Wafefunction: wavefunction object.
 
         """
-
-        raise NotImplementedError
+        self.number_of_circuits_run += 1
+        self.number_of_jobs_run += 1
```

Comment:
You've deleted `raise NotImplementedError`, now it will silently fail.
Also, it seems to me that perhaps it's worth adding this counter not inside the logic of the code, but rather as a decorator?

Reply:
I'm happy to reimplement this as a decorator but am actually wondering about when to use this style vs using a decorator.

Reply:
WDYT @dexter2206 ?

Reply:
I wouldn't recommend using a decorator here. `run_circuitset_and_measure` also increments those counters but by the number dependent on the input arguments and with some conditional logic, therefore it would be a bit tricky to write a single decorator that handles this. Even if we did that, the conditional logic of `run_circuitset_and_measure` would have to be repeated there. Or am I missing something?

Reply:
Sounds good if you say so 👍 
I'm in the phase where I would write all the code as decorators ;)